### PR TITLE
Allow modules to add custom product flags

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | title says it all
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | install this example module: https://www.dropbox.com/s/l0st1g41aerrt4c/kpflagsmodifier.zip?dl=0
| Sponsor company  | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11614)

### CODE REVIEW

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 1]
```diff
@@ -466,6 +467,11 @@ public function getFlags()
             );
         }
 
+        Hook::exec('productFlagsModifier', array(
```
Comment (mickaelandrieu) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]: is the name of the hook consistent with the others?

**Review by mickaelandrieu — COMMENTED** [review]
1 question

File: src/Adapter/Presenter/Product/ProductLazyArray.php (hunk 1 above)
Comment (kpodemski) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: i was wondering how to name this hook, we could have:
`actionProductFlagsModifier`

but it's not really any action

`displayProductFlagsModifier`

is more likely using to display something, not alter

so i decided to do this without prefix as some others hooks with same purpose, it should be it or `action` prefix, i am open for suggestions here

Comment (Quetzacoalt91) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: As far as I can tell, `*Modifier` hooks start with `action`. :)

Comment (mickaelandrieu) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: Let's go for `action` prefix, thanks @Quetzacoalt91 !

Comment (kpodemski) [reply src/Adapter/Presenter/Product/ProductLazyArray.php]: @mickaelandrieu done, do i need to squash those two commits or it is no longer needed in current workflow?

**Review by PierreRambaud — APPROVED** [review]

**Comment by marionf** [comment]
All good, thank you @kpodemski 

capture d ecran_753

**Comment by PierreRambaud** [comment]
Thanks @kpodemski and @marionf :)

**Comment by morghy1992** [comment]
@kpodemski thank you for this post. 
I just installed the module but could you explain how can enable it? And I would like to insert SOLD OUT when the products are 0 quantities ( I use attributes for my products).

**Comment by morghy1992** [comment]
> All good, thank you @kpodemski
> 
> capture d ecran_753

Hi, how did you insert the label CHEAP?

**Comment by davidhardik** [comment]
@kpodemski Hi, I am using 1.7.6.4 and I tried configuring but label is not visible on the page. Kindly let me know if I am missing something.

Thank you

**Comment by kpodemski** [comment]
@davidhardik could you share your code?

**Comment by davidhardik** [comment]
Hi, 

I am using your module only. The sample module that you have shared above.

**Comment by pelosek2525** [comment]
Hi , how can i modify custom flags and append them on the products? Thank you!

**Comment by kpodemski** [comment]
@pelosek2525 here's an example of such a module: https://github.com/PrestaShopSchool/kptopsellerslabel

**Pull request merged**